src/Network/Wai/SAML2/Config.hs

```diff
@@ -48,9 +49,18 @@ data SAML2Config = SAML2Config {
 -- SP's private key and @publicKey@ as the IdP's public key. You should
 -- almost certainly change the resulting settings.
 saml2Config :: PrivateKey -> PublicKey -> SAML2Config
-saml2Config privKey pubKey = SAML2Config{
+saml2Config privKey pubKey = (defaultSAML2Config pubKey){
+    saml2PrivateKey = Just privKey
+}
+{-# DEPRECATED saml2Config "use defaultSAML2Config instead" #-}
+
+-- | 'defaultSAML2Config' @publicKey@ constructs a 'SAML2Config' value
+-- with the most basic set of options possible using @publicKey@ as the
+-- IdP's public key. You should almost certainly change the resulting settings.
+defaultSAML2Config :: PublicKey -> SAML2Config
+defaultSAML2Config pubKey = SAML2Config{
```

Comment:
I wonder if this should be named differently, e.g. `saml2ConfigWithoutEncryption` or similar. I think the question here is what the library should encourage people to use. Personally, I think it's a good idea to nudge people towards using encrypted assertions if possible / make it evident when they are not using them.

Reply:
No, I don't think so; most IdPs doesn't even accept plain HTTP URI as an assertion consumer so confidentiality is assured in the transport layer. Some Identity providers such as Google Workspace don't even support encrypted assertions, possibly because it's redundant given TLS.

Reply:
I don't necessarily think that an additional encryption layer is redundant, because even if TLS is used, then the assertion itself may still get passed around internally on the SP end between different systems, stored, etc. and it is better to have it encrypted. At worst, the additional layer of encryption can't hurt (aside from a few extra CPU cycles). 

The main point for me is that the more secure option should be the default. I.e., if you don't want a particular security mechanism, you should always have to opt-out of it, rather than opt-in if you want it.

Reply:
> assertion itself may still get passed around internally on the SP end between different systems

How does it help to pass an encrypted assertion to the service provider if the service provider decrypts the assertion and store it somewhere else? 

"a few extra CPU cycles" is obviously exaggerated (RSA is not fast, hence the combination of RSA-OAEP and AES in EncryptedAssertion).

Also, in general SAML assertions do not contain sensitive data unless they really configure the IdP to do so.

Maintaining a key pair is a pain for system administrators; accidentally expiring a certificate could cause service disruption so I think it's even safer to reduce moving parts with respect to security.

Reply:
Thanks for making this change, especially if you did not agree with it! 

> How does it help to pass an encrypted assertion to the service provider if the service provider decrypts the assertion and store it somewhere else?

It does not prevent the SP from doing silly things, but it does give them the option to do something reasonable more easily.

> "a few extra CPU cycles" is obviously exaggerated 

Yep, indeed! 

> Also, in general SAML assertions do not contain sensitive data unless they really configure the IdP to do so.

I don't think this is true. SAML assertions often contain personal data (even something simple as someone's name, date of birth, etc.), which is always sensitive from a data privacy perspective.